lib/ht_sip_validator/validation/checksums.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+require "ht_sip_validator/validation/base"
+
+module HathiTrust
+  module Validation
+    module Checksums
+      # validates that package contains checksums
+      class Exists < HathiTrust::Validation::Base
+        def validate
+          unless @sip.files.include?(HathiTrust::SIP::CHECKSUM_FILE)
+            @messages << Message.new(
+              validator: self.class,
+              validation: :exists,
+              level: Message::ERROR,
+              human_message: "SIP is missing #{HathiTrust::SIP::CHECKSUM_FILE}",
+              extras: { filename: HathiTrust::SIP::CHECKSUM_FILE }
+            )
+          end
+
+          super
+        end
+      end
+
+      # validates that checksums exist for all files in the sip
+      class FileListComplete < HathiTrust::Validation::Base
+        def validate
+          @sip.files.each do |filename|
```

Comment:
Yup, that's the only case where there should be a file in the SIP package that doesn't have a checksum.

There may be some edge cases where there can be junk files like .DS_Store, Thumbs.db, or other files that specifically get ignored out of the SIP where it doesn't really matter if they have checksums or not -  but I think the right solution there to have the SIP object ignore those rather than to make a special provision in the checksum validator, and I'm OK not worrying about that for now.